### DISCUSSION TITLE
Limited user for security

### DISCUSSION THREAD

**Discussion by luke-jr** [arendst/Tasmota#20022] [Ideas]
I would like to setup a second username/password which has only limited access to devices, for example to toggle already-defined switches or read BLE reports - but NOT to update firmware or change settings.

Would this be complex to add?

**Comment by sfromis** [comment]
As Tasmota has no user management, it would not be a simple thing to add.

One thing you could do would be to have the device started in user mode, instead of admin. Then you can use MQTT (you need that) to send a command to switch to admin mode when relevant, do the updates, and then return to user mode. Or you can just do the updates via commands on MQTT, where all is allowed.

**Reply by luke-jr** [reply]
Just a second password with hardcoded limited access would suffice :/

**Comment by Jason2866** [comment]
Will not be implemented. Tasmota Web GUI is not designed not ment for this use case. Resources of the MCU / Flash are very limited.

**Reply by luke-jr** [reply]
It's not meant to be possible to secure it?

**Reply by Jason2866** [reply]
No not possible, only a feature that will not be considered to be implemented. You can add in your fork everything you want.

**Reply by sfromis** [reply]
I already suggested a strategy to deal with it. More generally, the rudimentary Tasmota web UI is not really meant as a "dashboard" for daily operation, which is expected to happen in a backend for multiple devices. Hence, the web UI is very low on dashboard-like features.
